kong/init.lua

```diff
@@ -347,57 +522,80 @@ function Kong.balancer()
     end
   end
 
-  current_try.ip   = addr.ip
-  current_try.port = addr.port
+  local ip, port = addr.ip, addr.port
+
+  current_try.ip   = ip
+  current_try.port = port
+
+  if mock then
+    ip, port = mock_addr()
+  end
 
   -- set the targets as resolved
   ngx_log(ngx_DEBUG, "setting address (try ", addr.try_count, "): ",
-                     addr.ip, ":", addr.port)
-  local ok, err = set_current_peer(addr.ip, addr.port)
+    ip, ":", port)
```

Comment:
style: this broke the indentation of these arguments, which was previously correct